src/Algebra/Properties/Semiring/Binomial.agda

```diff
@@ -0,0 +1,235 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The Binomial Theorem for *-commuting elements in a Semiring
+--
+-- Freely adapted from PR #1287 by Maciej Piechotka (@uzytkownik)
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles using (Semiring)
+open import Data.Bool.Base using (true)
+open import Data.Nat.Base as Nat hiding (_+_; _*_; _^_)
+open import Data.Nat.Combinatorics
+  using (_C_; nCn≡1; nC1≡n; nCk+nC[k+1]≡[n+1]C[k+1])
+open import Data.Nat.Properties as Nat
+  using (<⇒<ᵇ; n<1+n; n∸n≡0; +-∸-assoc)
+open import Data.Fin.Base as Fin
+  using (Fin; zero; suc; toℕ; fromℕ; inject₁)
+open import Data.Fin.Patterns using (0F)
+open import Data.Fin.Properties as Fin
+  using (toℕ<n; toℕ-fromℕ; toℕ-inject₁)
+open import Data.Fin.Relation.Unary.Top
+  using (view; ‵fromℕ; ‵inject₁; view-fromℕ; view-inject₁)
+open import Function.Base using (_∘_)
+open import Relation.Binary.PropositionalEquality.Core as ≡
+  using (_≡_; _≢_; cong)
+
+module Algebra.Properties.Semiring.Binomial
+  {a ℓ} (S : Semiring a ℓ)
+  (open Semiring S)
+  (x y : Carrier)
```

Comment:
Explicit or implicit for `x` and `y`?

Reply:
Ah, no... unless we are willing to make the definition in `CommutativeSemiring` slightly clunkier. Live and learn!

Reply:
Yup I think better to leave these explicit :+1: